[debops.docker] Rename role to debops.docker_server

imrejonk opened the pull request:
This renames the debops.docker role to debops.docker_server in preparation of adding a role that will provide client functionality like network and container management. As discussed with @drybjed.

I've tested my changes in a VM testing environment. Let me know if I did a good enough job :)

imrejonk committed:
[debops.docker_server] Role rename (WIP)

imrejonk committed:
[debops.docker_server] finish role rename

The debops.docker role has been renamed to debops.docker_server in
preparation of adding a role that will provide client functionality like
network and container management.

drybjed commented on CHANGELOG.rst:
```diff
@@ -583,7 +587,7 @@ Changed
   :command:`dnsmasq` or :command:`unbound`, Docker containers might have
   misconfigured DNS nameserver in :file:`/etc/resolv.conf` pointing to
-  ``127.0.0.1``. In these cases, the :ref:`debops.docker` role will configure
+  ``127.0.0.1``. In these cases, the debops.docker role will configure
```
You could change the reference to point to the new role. Changelog can be updated over time in cases like this. :-)

drybjed commented on ansible/roles/debops.docker_server/COPYRIGHT:
```diff
@@ -1,4 +1,4 @@
+debops.docker_server - Manage Docker server using Ansible
 
 Copyright (C) 2015-2016 Maciej Delmanowski <drybjed@gmail.com>
 Copyright (C) 2015-2017 DebOps https://debops.org/
```
You should add yourself here. And update the DebOps year as well.

drybjed commented on ansible/roles/debops.docker_server/defaults/main.yml:
```diff
@@ -13,119 +13,119 @@
+docker_server__distribution: '{{ ansible_local.core.distribution
                           if (ansible_local|d() and ansible_local.core|d() and
                               ansible_local.core.distribution|d())
                           else ansible_distribution }}'
```
It would look nicer if the values were indented to the new variable name lengths, here and in other such cases.

drybjed commented on ansible/roles/debops.docker_server/templates/etc/ansible/facts.d/docker.fact.j2:
```diff
@@ -9,7 +9,7 @@ import subprocess
 import os
```
You should rename the fact file to the `docker_server.fact.j2` as well, don't forget to update the task that templates it.

drybjed commented on docs/ansible/roles/debops.docker_server/upgrade.rst:
```diff
@@ -7,6 +7,13 @@ The upgrade notes only describe necessary changes that you might need to make
+
+The role has been renamed from ``debops.docker`` to ``debops.docker_server``.
+The variable prefix has been updated to ``docker_server__``. You will need to
+update your inventory to continue using this role.
```
This should be included in the `docs/news/upgrades.rst` file as well. The per-role upgrade files will be removed at some point. I would also mention that the hosts need to be moved to the new Ansible inventory group for the role to be enabled, because it's a very rare change.

drybjed reviewed (changes requested):
Nicely done! Can I assume that you only renamed everything, without any changes in functionality, etc.? If you found something, this should be done as a separate PR.

There are a few things I would change before merging this pull request. See the comments below.

drybjed commented:
You should also update the `.gitlab-ci.yml` file in the root of the DebOps repository so the `debops.docker_server` role is tested in the GitLab CI pipeline.

imrejonk committed:
[debops.docker_server] apply feedback from drybjed

- documentation fixes
- indentation
- rename fact file to docker_server.fact
- add debops.docker_server to GitLab CI pipeline

imrejonk commented:
Hey @drybjed, thanks for the feedback! I totally forgot about indentation there...
How's this?

imrejonk commented:
To clarify: these changes only rename the docker role, there are no changes in functionality.

drybjed reviewed (approved):
Nicely done, thanks! Let's see if the GitLab CI test breaks. :-)

imrejonk commented:
Woo! All green :man_dancing:
